Add Test DP-1.17: Egress traffic DSCP rewrite

desaimg1 opened the pull request:
Readme location:
https://github.com/openconfig/featureprofiles/blob/main/feature/qos/otg_tests/egress_traffic_classification_and_rewrite_test/README.md

Test is failing coz of this bug ID : [b/419618177](https://issuetracker.google.com/issues/419618177)

The log is attached here: https://partnerissuetracker.corp.google.com/issues/415458482

desaimg1 commented on feature/qos/otg_tests/egress_traffic_classification_and_rewrite_test/egress_traffic_classification_and_rewrite_test.go:
```diff
@@ -0,0 +1,1081 @@
+
+// Configures the given DUT interface.
+func configInterfaceDUT(p *ondatra.Port, a *attrs.Attributes, dut *ondatra.DUTDevice) *oc.Interface {
+	i := a.NewOCInterface(p.Name(), dut)
```
comments addressed

sezhang2 commented:
> > The qos config needs to use cfgplugins/qos.go.
> 
> Hi @dplore Thanks for reviewing it. we are using CLI after pushing partial supported config through OC and based on OC config we are adding the DSCP values which is specific to this testcase. In this case is it necessary to move the config to cfgpluggins.

Hi @desaimg1 Can you use a deviation for CLI config per @dplore 's comment and open a bug to track it? Then we can review/approve it. thanks!

desaimg1 commented:
> > > The qos config needs to use cfgplugins/qos.go.
> > 
> > 
> > Hi @dplore Thanks for reviewing it. we are using CLI after pushing partial supported config through OC and based on OC config we are adding the DSCP values which is specific to this testcase. In this case is it necessary to move the config to cfgpluggins.
> 
> Hi @desaimg1 Can you use a deviation for CLI config per @dplore 's comment and open a bug to track it? Then we can review/approve it. thanks!

HI @sezhang2 I have already added deviations and bug details in the script. updated the script now as well..

dplore commented on internal/cfgplugins/qos.go:
```diff
@@ -372,3 +375,53 @@ func GetPolicyCLICounters(t *testing.T, dut *ondatra.DUTDevice, policyName strin
+			aristaConfigureRemarkIpv4(t, dut)
+			aristaConfigureRemarkIpv6(t, dut)
+		}
+	}
```
Please include the OC standard way to rewrite/remark DSCP.  The key path is:

[/qos/classifiers/classifier/terms/term/actions/remark/config/set-dscp](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-qos.html#qos-classifiers-classifier-terms-term-actions-remark-config-set-dscp)

desaimg1 replied:
it is included in qos.go file in line number 156.
		if !deviations.QosRemarkOCUnsupported(dut) {
			action.GetOrCreateRemark().SetDscp = ygot.Uint8(class.RemarkDscp)
		}

since it is aborting the script, we are adding it as another function.

dplore replied:
ok, please add comments explaining this ordering of the configuration both inline where ConfigureQosDscpRemarkSpecific is called, and as a description for the function.

Also this should be a private function, so it should be lower case.

desaimg1 commented:
> The qos remarking function needs to specify the OC standard way to configure DSCP (in addition to the vendor deviation you discovered)

@dplore 
OC standard way of DSCP rewrite is already inluded in qos.go of cfgplugins file in line number 156.
if !deviations.QosRemarkOCUnsupported(dut) {
action.GetOrCreateRemark().SetDscp = ygot.Uint8(class.RemarkDscp)
}

since it is aborting the script, we are adding it as another function.

dplore commented:
Approved.  Please bring your branch up to date with main and this can be merged
